Source/Core/AudioCommon/OpenALStream.cpp

```diff
@@ -146,18 +146,39 @@ void OpenALStream::SoundLoop()
 	memset(uiBuffers, 0, numBuffers * sizeof(ALuint));
 	uiSource = 0;
 
+	// Checks if a X-Fi is being used. If it is, disable FLOAT32 support.
+	if (strstr(alGetString(AL_RENDERER), "X-Fi") != NULL)
+	{
+		float32_capable = false;
+	}
+
 	// Generate some AL Buffers for streaming
 	alGenBuffers(numBuffers, (ALuint *)uiBuffers);
 	// Generate a Source to playback the Buffers
 	alGenSources(1, &uiSource);
 
 	// Short Silence
-	memset(sampleBuffer, 0, OAL_MAX_SAMPLES * numBuffers * FRAME_SURROUND_FLOAT);
+	if (float32_capable)
+	{
+		memset(sampleBuffer, 0, OAL_MAX_SAMPLES * numBuffers * FRAME_SURROUND_FLOAT);
+	}
+	else
+	{
+		memset(sampleBuffer, 0, OAL_MAX_SAMPLES * numBuffers * FRAME_SURROUND_SHORT);
+	}
```

Comment:
If you only have one line inside each block, you can write it like this:

    if (float32_capable)
        memset(sampleBuffer, 0, OAL_MAX_SAMPLES * numBuffers * FRAME_SURROUND_FLOAT);
    else
        memset(sampleBuffer, 0, OAL_MAX_SAMPLES * numBuffers * FRAME_SURROUND_SHORT);

This also applies to lines 150-153 and 174-181. Don't do it to lines 286-297.